### PR TITLE
Upgrade the Gradle plugin to 0.10.2 and restrict the LDAP tests to non-Windows environments 

### PR DESCRIPTION
**Pull request by BuddhiWathsala**
## Purpose
Fixes https://github.com/ballerina-platform/ballerina-standard-library/issues/1655

More details about the solution https://github.com/ballerina-platform/ballerina-standard-library/issues/1655#issuecomment-888843331

## Checklist
- [x] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests

### CODE REVIEW

**Comment by dilanSachi** [comment]
Can we enable ldap tests for declarative auth also in this PR?
https://github.com/ballerina-platform/module-ballerina-grpc/blob/3b8323997ca3575eb3250f569b66b2f408b897a2/ballerina/tests/55_declarative_authentication_client.bal#L212

Need to insert ldap config here [also](https://github.com/ballerina-platform/module-ballerina-grpc/blob/3b8323997ca3575eb3250f569b66b2f408b897a2/ballerina/tests/55_declarative_authentication_service.bal#L109)

File: ballerina/tests/52_unary_ldap_auth_client.bal [hunk 1]
```diff
@@ -14,125 +14,140 @@
 // specific language governing permissions and limitations
 // under the License.
 
+import ballerina/jballerina.java;
 import ballerina/test;
 
 @test:Config {
-    enable: false,
+    enable: true,
```
Comment (lahirusamith) [inline ballerina/tests/52_unary_ldap_auth_client.bal:21]: Do we need to explicitly mention the `enable: true` here?

Comment (BuddhiWathsala) [reply ballerina/tests/52_unary_ldap_auth_client.bal:21]: This has been a practice in gRPC. We'll group those tests later.

**Comment by BuddhiWathsala** [comment]
@dilanSachi, in order to allow declarative LDAP tests we have to start auth enabled services as follows:
1. In Ubuntu we can start services as it is(enabling LDAP config)
2. But in Windows, we cannot start LDAP enabled services

So, we have to conditionally start this service. We could do that using `attach` and `start` APIs in the listener. But when I did that, [this lang level error](https://github.com/ballerina-platform/ballerina-lang/issues/31955) occurred. Therefore, I disabled the declarative ones until this issue is being fixed.

File: ballerina/tests/52_unary_ldap_auth_client.bal [hunk 2]
```diff
@@ -14,125 +14,140 @@
...
+        var response = helloWorldEp->testStringValueReturn(requestMessage);
+        if response is Error {
+            test:assertEquals(response.message(), "Permission denied");
+        } else {
+            test:assertFail(msg = "Expected grpc:Error not found.");
+        }
     }
 }
+
+isolated function isWindowsEnvironment() returns boolean = @java:Method {
```
Comment (dilanSachi) [inline ballerina/tests/52_unary_ldap_auth_client.bal:149]: Maybe we can put this function in `test_utils.bal`

**Review by dilanSachi — APPROVED** [review]

File: ballerina/tests/52_unary_ldap_auth_client.bal (hunk 2 above)
Comment (BuddhiWathsala) [reply ballerina/tests/52_unary_ldap_auth_client.bal]: Fixed in https://github.com/ballerina-platform/module-ballerina-grpc/pull/356/commits/0680b45f1f9b83d53fa5a2c0a933c4dec92af6fb

**Review by MadhukaHarith92 — APPROVED** [review]
LGTM

**Pull request merged**